ortho_seq_code/plotclass.py

```diff
@@ -0,0 +1,361 @@
+import os
+import numpy as np
+from ortho_seq_code.constants_orthoseqs import *
+import matplotlib.pyplot as plt
+import click
+from ortho_seq_code.utils import create_dir_if_not_exists
+
+
+class rf1d:
+    # Initialize rf1d object
+    def __init__(
+        self, ndarray, alphbt_input, molecule="protein", phenotype=None, out_dir=None
+    ):
+        try:
+            self.x = ndarray
+            self.x_flat = list(ndarray.flatten())
+            self.s = ndarray.shape[0]
+            self.d = ndarray.shape[1]
+            self.ind = np.arange(self.s)
+            self.num_dm = np.arange(self.d)
+            self.alphbt_input = alphbt_input.split(",")[0 : ndarray.shape[1]]
+            self.m = molecule
+            self.complist = ["<", ">", "<>", "><"]
+            self.phenotype = phenotype
+            self.out_dir = out_dir
+        except:
+            print(
+                "Error: Please provide a valid ndarray object and molecule type when initializing."
+            )
+
+    def summary(self):
+        print("rf1d Object:\n")
+        print("Number of sites:", str(self.s))
+        print("Number of dimensions:", str(self.d))
+        print("Alphabet input:", str(self.alphbt_input))
+        print("Molecule:", str(self.m) + "\n")
+        if self.phenotype is not None:
+            print("Phenotype represents", self.phenotype, "values")
+        if self.out_dir is not None:
+            print("Image output directory:", self.out_dir)
+        print("Highest rFon1D magnitudes:")
+        self.sort(by_magnitude=True)
+
+    # rFon1D bar plot
+    def barplot(
+        self,
+        xlab="Sequence Site",
+        title=None,
+        fixed_width=False,
+        include_borders=True,
+        out_dir=None,
+    ):
+        if any(i != 0 for i in self.x_flat):
+            if not fixed_width:
+                data_null = np.where(
+                    np.array(self.x_flat) == float(0), float("nan"), self.x_flat
+                )
+            else:
+                data_null = self.x_flat
+
+            # Constants/constant arrays
+            width = 1 / self.s
+            sd = self.s * self.d
+
+            # Re-vectorization with null values (converts to dict)
+            dim_num = dict()
+            for i in self.ind:
+                dim_num[i] = [
+                    data_null[j] for j in np.arange(self.d * i, self.d * i + self.d)
+                ]
+            # some_dim = [data_array_flat[i], i for i in range(0, 160, 4)]
+
+            # Remove all null data
+            # dim_loc is the indeces of all non-null data
```

Comment:
could this be used here instead? 

https://numpy.org/doc/stable/reference/generated/numpy.nonzero.html